common/persistence/visibility/store/elasticsearch/visibility_store.go

```diff
@@ -598,14 +600,22 @@ func (s *visibilityStore) buildSearchParameters(
 		params.SearchAfter = token.SearchAfter
 	}
 
-	if overStartTime {
-		params.Sorter = append(params.Sorter, elastic.NewFieldSort(searchattribute.StartTime).Desc())
+	if strings.HasPrefix(s.index, v2IndexPrefix) {
```

Comment:
Why do you need this `if`? Why not to sort by these 3 fields always?

Reply:
Technically not much difference. But I want to take a safer approach to not touch the existing code. So if anything bad happen we are sure that we rollback to existing logic.